okio/src/commonMain/kotlin/okio/Options.kt

```diff
@@ -71,10 +71,8 @@ class Options private constructor(
       val trieBytes = Buffer()
       buildTrieRecursive(node = trieBytes, byteStrings = list, indexes = indexes)
 
-      val trie = IntArray(trieBytes.intCount.toInt())
-      var i = 0
-      while (!trieBytes.exhausted()) {
-        trie[i++] = trieBytes.readInt()
+      val trie = IntArray(trieBytes.intCount.toInt()) {
+        trieBytes.readInt()
```

Comment:
Don't you still need to loop over it?

Reply:
This is a loop. It's called once for each position in the array.

Reply:
oooooh I misread that the previouew code wasn't doing that in the block passed to `IntArray`.